ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/objectivefunctionevaluator/OverloadEvaluator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.openrao.searchtreerao.commons.objectivefunctionevaluator;
+
+import com.powsybl.openrao.commons.Unit;
+import com.powsybl.openrao.data.cracapi.cnec.FlowCnec;
+import com.powsybl.openrao.searchtreerao.result.api.FlowResult;
+import com.powsybl.openrao.searchtreerao.result.api.RemedialActionActivationResult;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Thomas Bouquet {@literal <thomas.bouquet at rte-france.com>}
+ */
+public class OverloadEvaluator implements CostEvaluator {
+    private final Set<FlowCnec> flowCnecs;
+    private final Unit unit;
+    private final MarginEvaluator marginEvaluator;
+    private static final double OVERLOAD_PENALTY = 10000d; // TODO : set this in RAO parameters
```

Comment:
Did you want to do this in this PR or in a different one?

Reply:
I think I'll do it in a future PR because I don't know when the parameters refactoring will be ready

Reply:
besides it is not a big feature to add